specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2020-09-01-preview/environmentSpecification.json

```diff
@@ -0,0 +1,414 @@
+        ],
+        "operationId": "EnvironmentSpecificationVersions_List",
+        "x-ms-examples": {
+          "List Environment Specification Version.": {
+            "$ref": "./examples/listEnvironmentSpecificationVersions.json"
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        },
+        "parameters": [
+          {
+            "$ref": "machineLearningServices.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "machineLearningServices.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "machineLearningServices.json#/parameters/WorkspaceNameParameter"
+          },
+          {
+            "$ref": "machineLearningServices.json#/parameters/APIVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/EnvironmentNameParameter"
+          }
+        ],
+        "responses": {
+          "default": {
+            "description": "Error",
+            "schema": {
+              "$ref": "machineLearningServices.json#/definitions/MachineLearningServiceError"
+            }
+          },
+          "200": {
+            "description": "Success",
+            "schema": {
+              "$ref": "#/definitions/EnvironmentSpecificationVersionResourceArmPaginatedResult"
+            }
+          }
+        }
+      }
+    }
+  },
+  "parameters": {
+    "EnvironmentNameParameter": {
+      "in": "path",
+      "name": "environmentName",
+      "type": "string",
+      "description": "Environment Name identifier.",
+      "required": true,
+      "x-ms-parameter-location": "method"
+    },
+    "EnvironmentVersionParameter": {
+      "in": "path",
+      "name": "environmentVersion",
+      "type": "string",
+      "description": "Environment Version identifier.",
+      "required": true,
+      "x-ms-parameter-location": "method"
+    }
+  },
+  "definitions": {
+    "AssetPath": {
+      "description": "Details of an AssetUri.",
+      "required": [
+        "path"
+      ],
+      "type": "object",
+      "properties": {
+        "path": {
+          "description": "The path of file/directory.",
+          "pattern": "\\w",
+          "type": "string"
+        },
+        "isDirectory": {
+          "description": "Whether the path defines a directory or a single file.",
+          "type": "boolean"
+        }
+      }
+    },
+    "EnvironmentSpecificationVersion": {
+      "description": "Data Transfer Object (DTO) for an Execution Environment Specification. Name and\r\nversion should default to the AzureML-Minimal environment version tested with the\r\ncurrent SDK client version as this is a main use-case. Only supported properties\r\nlisted in EnvironmentSpecificationProperties are allowed. For specification types\r\nthat support additional context, AssetPaths may contain a single Blob URI entry\r\npointing to a snapshot of that context.",
+      "type": "object",
+      "properties": {
```

Comment:
Please include the "required": [] attribute showing what properties are required on PUT requests

Reply:
added